File: educamp.adesso.ccd/src/fizz_buzz/their_fizz_buzz/FizzBuzzPrinter.java
package fizz_buzz.their_fizz_buzz;

import java.util.Iterator;
import java.util.List;

public class FizzBuzzPrinter {
  static void printResult(List<String> result) {
    Iterator<String> i = result.iterator();
    while(i.hasNext()) {
      System.out.println(i.next());
    }
  }
}

File: src/main/kotlin/fizzbuzz/FizzBuzzPrinter.kt
package fizzbuzz

import fizzbuzz.printer.Printer

class FizzBuzzPrinter(
    private val fizzBuzzConverter: FizzBuzzConverter,
    private val printer: Printer
) {

    fun print() {
        for (i in 1..100) {
            printer.print(fizzBuzzConverter.convertNumber(i))
        }
    }
}